modules/gapi/src/backends/onnx/gonnxbackend.cpp

```diff
@@ -126,8 +126,14 @@ class ONNXCompiled {
              std::vector<cv::Mat>& outs);
 
     std::vector<std::string> in_names_without_const;
+
+    cv::gapi::onnx::WorkloadTypeOnnxPtr workload_type;
+    unsigned int workload_listener_id = 0;
+    void setWorkLoadType(const std::string &type);
 public:
     explicit ONNXCompiled(const gapi::onnx::detail::ParamDesc &pp);
+    ~ONNXCompiled();
+    void configureWorkloadType(cv::gapi::onnx::WorkloadTypeOnnxPtr workload);
```

Comment:
Might be we can rename it to `listenToWorkloadType()`, if it will not change ONNX workload type state after all the changes.